packages/date-picker/src/basic/date-table.vue

```diff
@@ -267,7 +267,7 @@
           classes.push('default');
         }
 
-        if (selectionMode === 'day' && (cell.type === 'normal' || cell.type === 'today') && this.cellMatchesDate(cell, this.date || this.value)) {
+        if (selectionMode === 'day' && (cell.type === 'normal' || cell.type === 'today') && this.cellMatchesDate(cell, this.value)) {
```

Comment:
Remove `this.date` so that the "is-selected" class is only determined by the persisted value of the component.